docs/community/developer-guide.md

```diff
@@ -23,6 +23,107 @@ Make sure you have read and understood the main CONTRIBUTING guide:
 
 https://github.com/Azure/aks-engine/blob/master/CONTRIBUTING.md
 
+## Build AKS Engine from Source
+
+### Docker Development Environment
+
+The easiest way to start hacking on AKS Engine is to use a Docker-based environment. If you already have Docker installed then you can get started with a few commands.
+
+* Windows (PowerShell): `.\scripts\devenv.ps1`
+* Linux/OSX (bash): `./scripts/devenv.sh`
```

Comment:
I wonder if we should mention `make dev` here? Personally it's what I use and it's pretty convenient